homeassistant/components/mqtt/config_flow.py

```diff
@@ -264,7 +329,29 @@ class PlatformField:
 
 COMMON_MQTT_FIELDS = {
     CONF_QOS: PlatformField(QOS_SELECTOR, False, valid_qos_schema, default=0),
-    CONF_RETAIN: PlatformField(BOOLEAN_SELECTOR, False, bool),
+}
+
+PLATFORM_ENTITY_FIELDS = {
+    Platform.NOTIFY.value: {},
+    Platform.SENSOR.value: {
+        CONF_DEVICE_CLASS: PlatformField(SENSOR_DEVICE_CLASS_SELECTOR, False, str),
```

Comment:
It's not easy to understand what the `False` does.

I'd suggest to make the `PlatformField` kwarg-only (that can happen in a separate PR though)

Reply:
I'll leave this one for a future PR

Reply:
See https://github.com/home-assistant/core/pull/141498